src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -488,6 +564,24 @@ public override void ExecuteSiteRecoveryCmdlet()
                         primaryNic = providerSpecificDetails.SelectedSourceNicId;
                     }
 
+                    if (this.RecoveryVmTag == null ||
+                     this.RecoveryVmTag.Count == 0)
+                    {
+                        recoveryVmTag = providerSpecificDetails.TargetVmTags;
+                    }
+
+                    if (this.RecoveryNicTag == null ||
+                       this.RecoveryNicTag.Count == 0)
+                    {
+                        recoveryNicTag = providerSpecificDetails.TargetNicTags;
+                    }
+
+                    if (this.DiskTag == null ||
+                      this.DiskTag.Count == 0)
+                    {
+                        diskTag = providerSpecificDetails.TargetManagedDiskTags;
+                    }
```

Comment:
>} [](start = 20, length = 1)

V2A edit also we had to throw error right

Reply:
Fixed it

---
In reply to: [602052721](https://github.com/AsrOneSdk/azure-powershell/pull/21#discussion_r602052721) [](ancestors = 602052721)